pandas/core/indexes/numeric.py

```diff
@@ -202,6 +210,12 @@ def _ensure_dtype(cls, dtype: Dtype | None) -> np.dtype | None:
         dtype = pandas_dtype(dtype)
         if not isinstance(dtype, np.dtype):
             raise TypeError(f"{dtype} not a numpy type")
+        if dtype == np.float16:
+            # float16 not supported (no indexing engine)
+            dtype = np.dtype(np.float32)
+        if dtype == "float16":
```

Comment:
why disallowing the string but not the type?

Reply:
Yes, I'll make a new commit to fix this.

Reply:
I’ve updated the PR.